src/doc.c

```diff
@@ -697,47 +719,52 @@ static void expandTemplateMixinComments(TemplateMixin *tm, Scope *sc)
 
 void emitMemberComments(ScopeDsymbol *sds, Scope *sc)
 {
+    if (!sds->members)
+        return;
+
     //printf("ScopeDsymbol::emitMemberComments() %s\n", toChars());
     OutBuffer *buf = sc->docbuf;
 
-    if (sds->members)
+    const char *m = "$(DDOC_MEMBERS ";
+    if (sds->isModule())
+        m = "$(DDOC_MODULE_MEMBERS ";
+    else if (sds->isClassDeclaration())
+        m = "$(DDOC_CLASS_MEMBERS ";
+    else if (sds->isStructDeclaration())
+        m = "$(DDOC_STRUCT_MEMBERS ";
+    else if (sds->isEnumDeclaration())
+        m = "$(DDOC_ENUM_MEMBERS ";
+    else if (sds->isTemplateDeclaration())
+        m = "$(DDOC_TEMPLATE_MEMBERS ";
```

Comment:
just a note on efficiency, this should be ordered in order of decreasing likelihood, i.e. struct and template should come first and module should come last etc.